data/crac-creation/crac-creator-fb-constraint/src/main/java/com/powsybl/openrao/data/craccreation/creator/fbconstraint/craccreator/OutageReader.java

```diff
@@ -92,8 +91,8 @@ private void interpretWithNetwork(UcteNetworkAnalyzer ucteNetworkHelper) {
                     this.isOutageValid = false;
                     this.invalidOutageReason = String.format("one of the two Xnodes of outage %s was not found in the network: %s, %s", outage.getId(), hvdcVH.getFrom(), hvdcVH.getTo());
                 } else {
-                    outageElementIds.add(dl1.getId());
-                    outageElementIds.add(dl2.getId());
+                    outageElementIdsAndContingencyType.put(dl1.getId(), ContingencyElement.of(dl1).getType());
```

Comment:
The idea here was to use "ContingencyElement.of" to no repeat elsewhere than in core, the information that a dangling line on network is associated to "danging line contingency type", in case this relation change in core in the future. But maybe the existence of the adder on rao side already create information duplication.

Reply:
@obrix  Related issue: I have a ContingencyElementType.BRANCH and an associated builder addBranch on core, but Branch had no implementation, it is always a subtype which is implemented in the network: Line, TieLine, 2WT. All these substypes have their own ContingencyElementType, and that's why BRANCH do not appear in  "ContingencyElement.of".
Question is : Should I remove ContingencyElementType.BRANCH from my adder in rao? (I guess answer is no but: should I remove addBranch and ContingencyElementType.BRANCH from core builder?)

Reply:
Ok, after discussing it directly lets keep it like this.